### ISSUE TITLE
Clean Up Repo (source code vs tests; branches) - executable package to pip!

### ISSUE THREAD

**Issue by przell** [Open-EO/openeo-udf-python-to-r#18]
see title

**Comment by m-mohr** [comment]
I created us a repository https://github.com/Open-EO/r4openeo-usecases where we can move our use cases to.

**Comment by m-mohr** [comment]
Tried to package this but miserably failed... It seems if you move files down into a tests folder then Python is too stupid to find the correct files (see dev channel).

**Comment by przell** [comment]
Let's find a python guy to do this for us! I'll ask around in our institute

**Comment by pierocampa** [comment]
hey, what kind of error are you getting?

**Comment by przell** [comment]
@pierocampa, thanks for offering your help 😃 
The last I heard of what is not working, @m-mohr, feel free to add or clarify: 
> ...How do I need to fix this file: <https://github.com/Open-EO/openeo-udf-python-to-r/blob/main/tests/test.py> So that when running it I don't get "no module named udf_lib"? I already tried dozens of variants, including from ..udf_lib import execute_udf, create_dummy_cube...

**Comment by pierocampa** [comment]
I'm creating the env locally, I'll get back to you when I have news, good news hopefully!

**Comment by pierocampa** [comment]
Ok, so here's the package I could upload to TestPyPI:
https://test.pypi.org/project/openeo-udf-python-to-r-PCAMPALANI/

I couldn't reproduce your error, @m-mohr, but that's what I did:

- move the source module into a structured _src/_ folder:
```
./src/
  + <packagename_PCAMPALANI>/
    + __init__.py
    + udf_lib.py
```
-  removed the `packages = udf_lib` option from the setup.cgf file (it will look for an `udf_lib` folder inside `src/`) otherwise.
- defined the version of the package in the `__init__.py` file: this is what is referenced by the version statement you put in `setup.cfg` (NOTE: I made the version up, but you can enable automatic version inferring with [this package](https://pypi.org/project/setuptools-scm/))

**Comment by pierocampa** [comment]
(`_PCAMPALANI` to be removed when going official PyPI :))

**Comment by pierocampa** [comment]
CORRIGE: if you want to get rid of the package name (whatever it is), and be able to run the imports as you do in your test script, just move/rename the `udf_lib.py` module to `__init__.py`.

The important thing is to have the _./src/udf_lib/_ folder structure.

**Comment by m-mohr** [comment]
Thank you @pierocampa.

I've done the following:
1. create src/openeo_r_udf folder
2. Add __init__.py with the version declaration
3. Moved udf_lib.py (and executor.R) to the new folder
4. Removed the packages option from the setup.cfg
5. Changed the import in tests/test.py to from openeo_r_udf.udf_lib ...
6. Try to run `python tests/test.py`

**Comment by pierocampa** [comment]
Hi @m-mohr,
what you did looks good: then, as you said, you need to actually install the package to make your tests work (otherwise make your _src/_ folder part of the ´PYTHONPATH´ system variable.

In order to test your actual release of TestPyPI, you need to run:
```
$ python -m pip install --index-url https://test.pypi.org/simple/ --no-deps openeo_r_udf-USERNAME==vX.Y.Z
```

being USERNAME the suffix you put (that is a PyPI recommendation not to clash with existing packages of other people), and optionally specifying a version `==vX.Y.Z` if that is needed.

**Comment by m-mohr** [comment]
Thanks, yes it all seems to work now.
Published successfully to pypi now: https://pypi.org/project/openeo-r-udf/0.1.0/

**Comment by m-mohr** [comment]
We still need to clean-up the branches at some point, but that's not so urgent...

**Comment by przell** [comment]
@m-mohr, 
- can we somehow reference the package in the github front page. Either under "packages" or in the readme. 
- Do we need update some of the files to install the lib from pip?

**Comment by pierocampa** [comment]
hi there, 
@clausmichele realized the R script in the Python sources is not automatically included in the release, and also there is a context-dependent path in the sources that is used to reference it, so:

- explicitly include the R script in the PyPI release with the `include_package_data` option set to `True`, or to make a more fine-grained selection:
```
# setup.cfg
...
[options.package_data]
mypkg =
    *.R
```
- use the `pkg_resources` library to correctly point to it:
```
import pkg_resources
...
executor_R_path = pkg_resources.resource_filename(__name__, 'executor.R')
```

That should work! 

----

REFS: https://setuptools.pypa.io/en/latest/userguide/datafiles.html

**Comment by m-mohr** [comment]
Thanks a lot @pierocampa , that is very helpful. I'll update the package and release.

**Comment by m-mohr** [comment]
Added the README and someone else already committed the other fixes :-)

**Comment by m-mohr** [comment]
Released v0.1.1: https://pypi.org/project/openeo-r-udf/

Hope it works now, didn't have the chance yet to test it.

**Comment by clausmichele** [comment]
> 

I can confirm that it works!
